### PR TITLE
Add mlflow logging

### PR DESCRIPTION
**Pull request by DrKomodo**

### CODE REVIEW

File: timm/train.py [hunk 1]
```diff
@@ -817,6 +832,8 @@ def train(config_path: str | None = None):
         )
         with open(os.path.join(output_dir, 'args.yaml'), 'w') as f:
             f.write(args_text)
+        mlflow.log_text(args_text, "config.yaml")
+        _log_params(args)
```
Comment (dzjkb) [inline timm/train.py:836]: I'm thinking these parameters could be something we'll handle outside of `pytorch-image-models`?

File: timm/train.py [hunk 2]
```diff
@@ -1107,6 +1129,8 @@ def _backward(_loss):
     return OrderedDict([('loss', losses_m.avg)])
 
 
+
```
Comment (dzjkb) [inline timm/train.py]: that's a lot of blank lines

File: timm/utils/checkpoint_saver.py [hunk 3]
```diff
@@ -93,6 +94,8 @@ def save_checkpoint(self, epoch, metric=None):
                 if os.path.exists(best_save_path):
                     os.unlink(best_save_path)
                 os.link(last_save_path, best_save_path)
+                mlflow.pytorch.log_model(self.model, "model")
+
```
Comment (dzjkb) [inline timm/utils/checkpoint_saver.py:98]: more blank lines what's up with that

**Review by dzjkb — CHANGES_REQUESTED** [review]

File: timm/data/readers/reader_image_folder.py [hunk 4]
```diff
@@ -44,6 +46,8 @@ def find_images_and_targets(
             base, ext = os.path.splitext(f)
             if ext.lower() in types:
                 filenames.append(os.path.join(root, f))
+                if labels_df is not None:
+                    label = labels_df.loc[f]["label"]
```
Comment (dzjkb) [inline timm/data/readers/reader_image_folder.py]: you can do `labels_df.loc[f, "label"]`

**Review by dzjkb — CHANGES_REQUESTED** [review]

File: timm/train.py (hunk 1 above)
Comment (DrKomodo) [reply timm/train.py:836]: I'd rather log them here because the config parameters and their default values are defined here

**Review by dzjkb — APPROVED** [review]
